### PR TITLE
[Mailer] Do not use TLS by default for host !== 'localhost'

### LINKED ISSUE #34169: [Mailer][BC break] SSL is now assumed ON when host != localhost
**Symfony version(s) affected**: 4.4-dev

**Description**  

I'm using docker with a dev mailer: https://danfarrelly.nyc/MailDev/
The SMTP server is on port `25` without TLS and on host `mailedev`

My config:
```yaml
framework:
    mailer:
        dsn: 'smtp://maildev'
```

With Symfony 4.3, everything is OK
With Symfony 4.4, I got two issues:

1. The first one is that Symfony now think i'm using TLS. This is due to [this change](https://github.com/symfony/symfony/commit/5b8c4676d05987ce5346fdbf44fcac6654bc39c1#diff-fa745155c38f627fdc9e177a60da7570R52) I guess.

2. Then, even if I force `$tls = false` on line 54/55, something is still broken:
    ```
    Warning: stream_socket_enable_crypto(): SSL operation failed with code 1. OpenSSL Error messages:
    error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed
    ```

    And, it's "funny" because this crash the maildev daemon :boom: 

**Possible Solution**  

1. Do not force TLS when it is not specified
2. I don't know

**Additional context**  

I can create a reproducer if you want or I can work on it, but I will need some advice

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #34169
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by lyrixx** [comment]
OK, I validate the second commit is important.
If I put the following config: `mailer_dsn: "smtp://maildev:25"` where I hardcode the port, TLS is well disabled, but I still hit the following bug:
```
Warning: stream_socket_enable_crypto(): SSL operation failed with code 1. OpenSSL Error messages:
error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed
```

File: src/Symfony/Component/Mailer/Transport/Smtp/EsmtpTransport.php [hunk 1]
```diff
@@ -106,7 +102,7 @@ protected function doHeloCommand(): void
 
         /** @var SocketStream $stream */
         $stream = $this->getStream();
-        if (!$stream->isTLS() && \defined('OPENSSL_VERSION_NUMBER') && \array_key_exists('STARTTLS', $capabilities)) {
+        if ($stream->isTLS() && \defined('OPENSSL_VERSION_NUMBER') && \array_key_exists('STARTTLS', $capabilities)) {
```
Comment (fabpot) [inline src/Symfony/Component/Mailer/Transport/Smtp/EsmtpTransport.php]: This change is wrong and must be reverted. If TLS is disabled AND the server supports STARTTLS, then we enable it.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Mailer/Transport/Smtp/EsmtpTransport.php (hunk 1 above)
Comment (lyrixx) [reply src/Symfony/Component/Mailer/Transport/Smtp/EsmtpTransport.php]: I don't know how, But without this patch, it makes the server crash. 
How could I disable TLS? (if I know the server does not support it)

Comment (lyrixx) [reply src/Symfony/Component/Mailer/Transport/Smtp/EsmtpTransport.php]: I really don't know enough about TLS. Do you want a simple reproducer?
(I can work a bit on this topic too)

**Comment by stof** [comment]
Based on my investigation in #34242, it looks like the cause is that maildev is a buggy smtp server implementation, rather than being a bug in symfony.

**Comment by stof** [comment]
IMO, using the TLS port as the default port for non-localhost when unspecified is a good default. It means preferring TLS over unencrypted connections (potentially upgraded later thanks to STARTTLS but that can be blocked by a MitM discarding the STARTTLS capability).
If you want to force running on port 25 rather than 465, you can already do it by passing port 25 explicitly (which you do in your example btw, so this code is not even reached for the error you face).

so I vote -1 on this PR.

**Comment by lyrixx** [comment]
You are right :+1: Let's close this

**Pull request closed without merging**